src/compiler/msvc.rs

```diff
@@ -493,26 +557,60 @@ fn generate_compile_commands(executable: &Path,
         parsed_args.input.clone().into(),
         fo,
     ];
+    arguments.extend(parsed_args.preprocessor_args.clone());
     arguments.extend(parsed_args.common_args.clone());
-    Ok((CompileCommand {
+
+    let command = CompileCommand {
         executable: executable.to_owned(),
         arguments: arguments,
         env_vars: env_vars.to_owned(),
         cwd: cwd.to_owned(),
-    }, None, cacheable))
+    };
+
+    let dist_command = (|| {
+        // http://releases.llvm.org/6.0.0/tools/clang/docs/UsersManual.html#clang-cl
+        // TODO: Use /T... for language?
```

Comment:
This would be a useful followup. It looks like you could just replace `-c` with `-Tc` or `-Tp`:
https://docs.microsoft.com/en-us/cpp/build/reference/tc-tp-tc-tp-specify-source-file-type